File: back-end/src/main/java/br/com/pessoa/service/PessoaService.java
package br.com.pessoa.service;

import br.com.pessoa.model.Pessoa;

import java.util.List;

public interface PessoaService {

    List<Pessoa> listar();

    Pessoa findById(Long id);

    Pessoa cadastrar(Pessoa pessoa);

    void deletar(Long id);

    Pessoa atualizar(Long id, Pessoa pessoaAtualizada);

    Pessoa buscarPorCpf(String cpf);
}

File: src/app/pessoa.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Pessoa } from './pessoa.model';

@Injectable({
  providedIn: 'root'
})
export class PessoaService {
  private apiUrl = 'http://localhost:8080/pessoas';

  constructor(private http: HttpClient) { }

  listar(): Observable<Pessoa[]> {
    return this.http.get<Pessoa[]>(this.apiUrl);
  }

  salvar(pessoa: Pessoa): Observable<Pessoa> {
    return this.http.post<Pessoa>(this.apiUrl, pessoa);
  }

  deletar(id: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/${id}`);
  }

  atualizar(pessoa: Pessoa): Observable<Pessoa> {
    return this.http.put<Pessoa>(`${this.apiUrl}/${pessoa.id}`, pessoa);
  }

  buscarPorId(id: number): Observable<Pessoa> {
    return this.http.get<Pessoa>(`${this.apiUrl}/${id}`);
  }

  buscarPorCpf(cpf: string): Observable<Pessoa> {
    return this.http.get<Pessoa>(`${this.apiUrl}/cpf/${cpf}`);
  }
}
